feat(NgRoute): allow resolve promises on routeUpdate event

ivomju opened the pull request:
When using the reloadOnSearch defined as false, if you have a resolve property for that path the promise will not be invoked. My pull request is to enable that functionality, in our product we have a scenario where we need this feature I hope that this could help somebody else

ivomju committed:
feat(NgRoute): Update variable name from localsRouteIsUpdate to resolvePromises

d0g commented:
Can you check again for CLA? I've made a little mistake but if you check again it will be ok.

d0g commented:
@petebacondarwin can you give me some input about this?

d0g commented:
@IgorMinar can someone give me some input about this?

petebacondarwin commented on test/ngRoute/routeSpec.js:
```diff
@@ -1063,7 +1063,8 @@ describe('$route', function() {
 
-    it('should not reload a route when reloadOnSearch is disabled and only .search() changes', function() {
+    it('should not reload a route when reloadOnSearch is disabled and' +
+      'only .search() changes without resolve promises defined', function() {
```
Is this change to the test description correct?

d0g replied:
Yes, here you test without any resolve promises

petebacondarwin commented on src/ngRoute/route.js:
```diff
@@ -551,14 +551,29 @@ function $RouteProvider() {
     }
 
+    function buildResolvePromises(resolve) {
+      var locals = angular.extend({}, resolve);
```
I would call `locals` `resolvePromises` to keep it consistent with the name of the function.

petebacondarwin commented on src/ngRoute/route.js:
```diff
@@ -577,14 +592,9 @@ function $RouteProvider() {
           then(function() {
             if (nextRoute) {
-              var locals = angular.extend({}, nextRoute.resolve),
+              var locals = buildResolvePromises(nextRoute.resolve),
```
We could even rename `locals` here too

petebacondarwin commented:
I haven't tried running the tests but on the surface this looks right

ivomju committed:
feat(NgRoute): Update variables names

d0g commented:
@petebacondarwin I've updated what you commented. Thanks for the input about this

d0g commented:
@petebacondarwin Do you see this being accepted?

petebacondarwin commented:
Yes but not till after 1.5 is released now

d0g commented:
Ok, thanks mate ;)

ivomju committed:
fix($compile): removing unnecessary white space

When creating DOM commentary for ng-switch-when theres an unecessary white space at the end.

ivomju committed:
fix($compile): Remove spaces

ivomju committed:
feat(NgRoute): Revert changes

ivomju committed:
feat(NgRoute): Update implementation

ivomirra commented:
Hi guy's, can you estimate in witch version this will be released?

petebacondarwin commented:
@ivomirra - I will add it to our next meeting for discussion. I am slightly concerned about the change to async from sync code.

ivomirra commented:
Thanks @petebacondarwin. Can you give me feedback when you have a decision regarding this? Cheers

petebacondarwin commented:
will do

petebacondarwin commented:
Hi @ivomju we discussed this PR in our meeting today. The general feeling was that this didn't seem to fit with how the library currently works. What is the point of triggering the resolves again? We can't "reinject" them into the constructor, since we do not recreate the controller in this case.
Unless you can provide a really strong use case that is likely to impact a number of developers, we are not likely to accept this PR.

Narretz commented:
I'm going to close this issue because we haven't got any feedback. Leave a comment if you can provide new feedback.
